### PR TITLE
Task view 

### PR DESCRIPTION
**Pull request by alex6296**

### CODE REVIEW

File: src/default.css [hunk 1]
```diff
@@ -111,13 +112,36 @@ button {
...
   border-color: var(--primary-dark);
 }
 
+input[type="button"]:hover,
+button:hover
+{
+  background: var(--light);
+  color: var(--primary-dark);
+  border-color: var(--primary-dark);
+}
```
Comment (kvartborg) [inline src/default.css]: move the `input[type="button"]:hover` to the above definition and delete this css block its redundant.

File: src/api/task.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import request from './request'
+
+export default {
```
Comment (kvartborg) [inline src/api/task.js]: This file should be replaced by the one in master

File: src/views/Task/View.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  /**
+   *loads all tasks from the db into the state
+  */
+  loadTasks = async list=>{
+    const res = await Task.getSpecific(this.props.match.params.id);
+    console.log(res);
+
+    this.setState({tasks:res})
+    }
```
Comment (kvartborg) [inline src/views/Task/View.js]:
```suggestion
  }
```

File: src/views/Task/overView.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import { Link } from 'react-router-dom';
+
+export default class OverView extends Component{
+
+  constructor(props){
+    super(props);
+
+    this.state = {tasks:[]}
+
+   this.loadTasks();
```
Comment (kvartborg) [inline src/views/Task/overView.js]:
```suggestion
    this.loadTasks();
```

File: src/views/Task/overView.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+   * render a task
+   * @param {int} id
+   * @param {DateTime} start
+   * @param {DateTime} deadline
+   * @param {String} Description
+   * @param {String} Urgency
+   * @return {JsX}  a task as a list item
+   */
+  fieldRender(task){
+
```
Comment (kvartborg) [inline src/views/Task/overView.js]:
```suggestion
```

File: src/views/Task/overView.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      </Link>
+    )
+  }
+
+    /**
+   * Creates the Task overview view
+   * @return {JSX} View
+   */
+  render(){
+   // console.log(this.state.tasks)
```
Comment (kvartborg) [inline src/views/Task/overView.js]: Remove comment

File: src/views/Task/overView.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            <p>{task.start} → {task.deadline}</p>
+            <p>{task.urgency}</p>
+            <p>{task.description}</p>
+          </label>
+        </li>
+      </Link>
+    )
+  }
+
+    /**
```
Comment (kvartborg) [inline src/views/Task/overView.js]:
```suggestion
  /**
```

File: src/views/Task/View.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            <p><b>Task Id:</b> {task.id}</p>
+            <p>{task.start} → {task.deadline}</p>
+            <p>{task.urgency}</p>
+            <p>{task.description}</p>
+          </label>
+        </li>
+    )
+  }
+
+    /**
```
Comment (kvartborg) [inline src/views/Task/View.js]:
```suggestion
  /**
```

File: src/views/Task/View.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    super(props);
+
+    this.state = {tasks:[]}
+
+   this.loadTasks();
+  };
+
+  /**
+   *loads all tasks from the db into the state
+  */
```
Comment (kvartborg) [inline src/views/Task/View.js]: Indentation

File: src/views/Task/View.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+import { Link } from 'react-router-dom';
+
+export default class View extends Component{
+
+  constructor(props){
+    super(props);
+
+    this.state = {tasks:[]}
+
+   this.loadTasks();
```
Comment (kvartborg) [inline src/views/Task/View.js]: Indentation

File: src/default.css [hunk 11]
```diff
@@ -147,7 +148,20 @@ button.secondary:hover {
 }
 
 input.valid{
-  border-color : var(--success);
+  border-color : var(--succes);
```
Comment (kvartborg) [inline src/default.css]:
```suggestion
  border-color : var(--success);
```

File: src/Routes.js [hunk 12]
```diff
@@ -5,22 +5,18 @@ import Login from "./views/Login";
...
-    <div>
-      <Route path="/" exact component={Dashboard} />
-      <Route path="/login" component={Login} />
-      <Route path='/sign-up' component={SignUp}/>
-      <Route path="/task/create" component={Create} />
-      <Route path="/task/update" component={Update} />
-      <Route path="/logout" component={logout} />
-    </div>
-  </Router>
-);
```
Comment (kvartborg) [inline src/Routes.js]: This shouldn't be removed

**Comment by alex6296** [comment]
i can see i don't have any output incase of no task

File: src/Routes.js [hunk 13]
```diff
@@ -20,6 +22,8 @@ export default () => (
       <Route path='/sign-up' component={SignUp}/>
       <Route path="/task/create" component={Create} />
       <Route path="/task/update" component={Update} />
+      <Route path='/task/list' component={OverView} />,
```
Comment (kvartborg) [inline src/Routes.js]: Maybe we should call the component `OverView` `List` instead, it is easier to reason about if the route matches the component name.

File: src/views/Task/View.js [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+    this.loadTasks();
+  };
+
+  /**
+   *loads all tasks from the db into the state
+   */
+  loadTasks = async list=>{
+    const res = await Task.get(this.props.match.params.id);
+    console.log(res);
```
Comment (kvartborg) [inline src/views/Task/View.js]: Remove console.log

File: src/views/Task/View.js [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+  render(){
+    return (
+      <Layout>
+        <h1>Task overview</h1>
+        <ul>
+          {this.fieldRender(this.state.tasks)}
+        </ul>
+        <input type="button" value="Make Offer"/>
+        <input type="button" value="Chat"/>
+        <Link to='../List'>
```
Comment (kvartborg) [inline src/views/Task/View.js]: Relative paths in url is not a good solution, should be absolute.

File: src/views/Task/List.js [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+
+    this.state = {tasks:[]}
+
+    this.loadTasks();
+
+  };
+
+  /**
+   * loads all tasks from the db into the state
+   */
```
Comment (kvartborg) [inline src/views/Task/List.js]: Docblock missing params and return definition.

File: src/views/Task/TaskView.css [hunk 17]
```diff
@@ -0,0 +1,10 @@
+.view {
+  height: 100vh;
+  padding : 5%;
+  align-items: center;
+  justify-content: center;
+}
+
+.View form h1 {
```
Comment (kvartborg) [inline src/views/Task/TaskView.css]: Lower case ‘.View’ so its consistent.

File: src/views/Task/View.js [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+    super(props);
+
+    this.state = {tasks:[]}
+
+    this.loadTasks();
+  };
+
+  /**
+   *loads all tasks from the db into the state
+   */
```
Comment (kvartborg) [inline src/views/Task/View.js]: Missing params and return definition in docblock.

File: src/Routes.js [hunk 19]
```diff
@@ -20,6 +22,8 @@ export default () => (
       <Route path='/sign-up' component={SignUp}/>
       <Route path="/task/create" component={Create} />
       <Route path="/task/update" component={Update} />
+      <Route path='/task/list' component={List} />,
```
Comment (grammeaway) [inline src/Routes.js]: Might be more @kvartborg's area of expertise, but the routes might not be ideally named

File: src/default.css [hunk 20]
```diff
@@ -119,6 +119,7 @@ button {
 }
 
 input[type="submit"],
+input[type="button"],
```
Comment (grammeaway) [inline src/default.css]: input[type=button] shouldn't be necesarry, pretty sure they do roughly the same (correct me if I'm wrong)

File: src/default.css [hunk 21]
```diff
@@ -141,6 +142,7 @@ button.secondary {
 }
 
 input[type="submit"]:hover,
+input[type="button"]:hover,
```
Comment (grammeaway) [inline src/default.css]: same with the input button

File: src/views/Task/List.js [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+    )
+  }
+
+  /**
+   * Creates the Task overview view
+   * @return {JSX} View
+   */
+  render(){
+    return (
+      <Layout>
```
Comment (grammeaway) [inline src/views/Task/List.js]: Needs to be wrapped in a  <section> tag 😄

File: src/views/Task/TaskView.css [hunk 23]
```diff
@@ -0,0 +1,10 @@
+.view {
+  height: 100vh;
+  padding : 5%;
+  align-items: center;
+  justify-content: center;
+}
+
```
Comment (grammeaway) [inline src/views/Task/TaskView.css]: Try doing the above-mentioned section tag, don't think you need the component-specific CSS 👍

File: src/views/Task/View.js [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  /**
+   * Creates the Task overview view
+   * @return {JSX} View
+   */
+  render(){
+    return (
+      <Layout>
+        <h1>Task view</h1>
```
Comment (grammeaway) [inline src/views/Task/View.js]: wrap in <section> tags here a well ⛱

File: src/views/Task/List.js (hunk 22 above)
Comment (grammeaway) [reply src/views/Task/List.js]: \<section tags, fucking karkdown reacting to the tag 😆

File: src/views/Task/View.js (hunk 24 above)
Comment (grammeaway) [reply src/views/Task/View.js]: \<section> tags, fucking markdown reacting to the tag 😆

File: src/default.css (hunk 20 above)
Comment (alex6296) [reply src/default.css]: en button af typen <input> er det samme som <button>, den kan bare ikke tage billeder som value lige som en button kan

File: src/views/Task/View.js [hunk 25]
```diff
@@ -0,0 +1,64 @@
...
+        <ul>
+          {this.fieldRender(this.state.tasks)}
+        </ul>
+        <input type="button" value="Make Offer"/>
+        <input type="button" value="Chat"/>
```
Comment (kvartborg) [inline src/views/Task/View.js]:
```suggestion
        <button>Chat</button>
```

File: src/views/Task/View.js [hunk 26]
```diff
@@ -0,0 +1,64 @@
...
+   * @return {JSX} View
+   */
+  render(){
+    return (
+      <Layout>
+        <h1>Task view</h1>
+        <ul>
+          {this.fieldRender(this.state.tasks)}
+        </ul>
+        <input type="button" value="Make Offer"/>
```
Comment (kvartborg) [inline src/views/Task/View.js]:
```suggestion
        <button>Make Offer</button>
```

File: src/default.css (hunk 20 above)
Comment (alex6296) [reply src/default.css]: nvm min info er alt for gammel

File: src/Routes.js (hunk 19 above)
Comment (kvartborg) [reply src/Routes.js]: Do not have a better name so lets keep it for now.

**Pull request merged**